keps/sig-apps/2804-consolidate-workload-controllers-status/README.md

```diff
@@ -0,0 +1,912 @@
+| StatefulSet                        | -           | -         | -               | -         | -      | -        |
+| DaemonSet                          | -           | -         | -               | -         | -      | -        |
+| Job                                | -           | -         | -               | True / False when suspended / resumed | failed execution  (BackoffLimitExceeded, DeadlineExceeded)| completed execution |
+| CronJob**                          | -           | -         | -               | -         | -      | -        |
+
+**\* Success of the rollout is instead represented by a Progressing condition (status and reason)**
+
+**\*\* CronJob does not even have Conditions field in its Status**
+
+### Proposed Conditions
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+We are proposing 4 new conditions to be added to the following controllers:
+- Available (DaemonSet, ReplicaSet & ReplicationController, StatefulSet)
+- Progressing (DaemonSet, StatefulSet)
+- Waiting (Job)
+- Running (Job)
+
+The definitions for Progressing condition (including Failed/Complete) are similar to what we have for [Deployment controller](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#deployment-status).
+
+
+The following table is indicating what conditions are currently available (`X`) and what conditions will be added (`A`).
+
+|                                    | Waiting | Running | Progressing | Available |  ReplicaFailure | Suspended | Failed | Complete |
+| ---------------------------------  | --------| ------- | ----------- | --------- | --------------- | --------- | ------ | -------- |
+| ReplicaSet & ReplicationController | -       | -       | -           | A         | X               | -         | -      | -        |
+| Deployment                         | -       | -       | X           | X         | X               | -         | -      | -        |
+| StatefulSet                        | -       | -       | A           | A         | -               | -         | -      | -        |
+| DaemonSet                          | -       | -       | A           | A         | -               | -         | -      | -        |
+| Job                                | A       | A       | -           | -         | -               | X         | X      | X        |
+| CronJob                            | -       | -       | -           | -         | -               | -         | -      | -        |
+
+#### Progressing
+Kubernetes marks a DaemonSet or Stateful as `progressing` when:
+- The DaemonSet or StatefulSet is created
+- The DaemonSet or StatefulSet is scaling up or scaling down
+- New DaemonSet or StatefulSet pods become Ready or available
```

Comment:
RC/RS when batching replicas or when waiting for `MinReadySeconds`.

Reply:
please see https://github.com/kubernetes/enhancements/pull/2833#discussion_r796620901